src/components/media/controls/controls.tsx

```diff
@@ -0,0 +1,299 @@
+/* eslint-disable no-magic-numbers */
+import './controls.css'
+
+import Forward10RoundedIcon from '@mui/icons-material/Forward10Rounded'
+import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown'
+import KeyboardArrowUpIcon from '@mui/icons-material/KeyboardArrowUp'
+import PauseCircleFilledIcon from '@mui/icons-material/PauseCircleFilled'
+import PlayCircleFilledIcon from '@mui/icons-material/PlayCircleFilled'
+import Replay10RoundedIcon from '@mui/icons-material/Replay10Rounded'
+import VolumeOffRoundedIcon from '@mui/icons-material/VolumeOffRounded'
+import VolumeUpRoundedIcon from '@mui/icons-material/VolumeUpRounded'
+import Box from '@mui/material/Box'
+import Button from '@mui/material/Button'
+import IconButton from '@mui/material/IconButton'
+import Slider from '@mui/material/Slider'
+import Typography from '@mui/material/Typography'
+import React, { useEffect, useState } from 'react'
+
+import { formatDurationTime } from '../../helper'
+
+interface MediaIconButtonProps {
+  onClick: () => void
+  ariaLabel: string
+  icon: React.ReactNode
+  className?: string
+  dataCy?: string
+}
+
+export function MediaIconButton(props: MediaIconButtonProps) {
+  return (
+    <IconButton
+      onClick={props.onClick}
+      aria-label={props.ariaLabel}
+      className={props.className}
+      data-cy={props.dataCy}
+    >
+      {props.icon}
+    </IconButton>
+  )
+}
+
+interface MediaControls {
+  mediaElement: HTMLMediaElement
+}
+
+export function ProgressSlider(props: MediaControls) {
+  const formattedElapsedTime = formatDurationTime(
+    props.mediaElement.currentTime
+  )
+
+  function handleTimeChange(event: Event, newValue: number | number[]) {
+    props.mediaElement.currentTime = newValue as number
+  }
+
+  function renderBufferedProgressBar() {
+    const buffered = props.mediaElement.buffered
+    const ranges = []
+    for (let i = 0; i < buffered.length; i++) {
+      ranges.push({
+        start: buffered.start(i),
+        end: buffered.end(i),
+      })
+    }
+
+    return ranges.map((range, index) => (
+      <Box
+        key={index}
+        className="rustic-progress-buffered"
+        sx={{
+          backgroundColor: 'primary.light',
+          // eslint-disable-next-line no-magic-numbers
+          width: `${((range.end - range.start) / props.mediaElement.duration) * 100}%`,
+        }}
+      />
+    ))
```

Comment:
I dont think we need a container for each entry in TimeRanges. Just one should be sufficient. Its width can be computed using `TimeRanges.start()` and `TimeRanges.end()`

I also think that using MUI LinearProgress component with determinate variant is a better fit for this. 

References
1. https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/buffered
2. https://developer.mozilla.org/en-US/docs/Web/API/TimeRanges#normalized_timeranges_objects
3. https://mui.com/material-ui/react-progress/#linear-determinate